base/channels.jl

```diff
@@ -26,10 +26,21 @@ type Channel{T} <: AbstractChannel
         end
         new(Condition(), Condition(), :open, Array{T}(0), sz, Array{Condition}(0))
     end
+
+    # deprecated empty constructor
+    function Channel()
+        depwarn(string("The empty constructor Channel() is deprecated. ",
```

Comment:
How?

`Channel() = Channel{Any}(32)` is fine.
`Channel{T}() = Channel{T}(32)` is not callable.

```
julia> type Foo{T}
         foo::T
       end

julia> Foo{T}() = Foo{T}(1)
WARNING: static parameter T does not occur in signature for Type at REPL[2]:1.
The method will not be callable.
Foo{T}
```